Migrations

tchell opened the pull request:
OK, this should encompass all the changes needed for our updated database design.

- User class now has two static methods to load users, one to get the currently logged in user and one to get another user, as well as a way to write users.
- Book/Request/Loan has methods to write and delete themselves.

closes #64

tchell committed:
added User writeable

- User instance objects now have a method to write themselves to the correct
db location
- FirebaseAuth uid is now an attribute of User objects.

tchell committed:
added Book writeable and User instance loader

- Added a write method to Book so Books can write themselves to the
database
- Added a static getInstance method for User so that it is easy to load
the current user's info
- removed borrower from Book
- removed name from User
- refactored a couple tests
- fixed missing edit profile button closes #61

tchell committed:
Added Request writeble and delete.

- added methods to request for writing request objects.
- fixed the Request tests.

tchell committed:
Added Loan class with writable and tests

- #64

tchell committed:
refactored some database reads

- Books now have a delete method
- User now has a static method to load a user from a given username
- #64

tchell committed:
Refactored Loan/Requests fixed a query

- refactored the Loan and Request writeToDb to write to explicit
subchildren for each type i.e. "owner" or "requester"
- fixed the query in MyBookRecyclerViewAdapter that gets the borrower of
borrowed books
- #64

tchell committed:
removed unneeded queries

NErgezinger reviewed (dismissed):
Looks like everything's working except searching, which I fixed in the Request branch.

tchell committed:
fixed loading previous users

- Fixed a bug that that caused the previous user to be loaded when a new
user signed in.

tchell commented:
> Looks like everything's working except searching, which I fixed in the Request branch.

Yeah I didn't touch that query since I knew you fixed it.
